sway-lib-core/src/primitives.sw

```diff
@@ -16,6 +25,144 @@ impl u64 {
     pub fn bits() -> u32 {
         64
     }
+
+    pub fn to_le_bytes(self) -> [u8; 8] {
+        let output = [0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8];
+
+        asm(input: self, off: OFF, i: I, j: J, k: K, l: L, m: M, n: N, n: N, o: O, output: output, r1) {
+            and  r1 input off;
+            sw  output r1 i0;
+
+            srl  r1 input i;
+            and  r1 r1 off;
+            sw  output r1 i1;
+
+            srl  r1 input j;
+            and  r1 r1 off;
+            sw  output r1 i2;
+
+            srl  r1 input k;
+            and  r1 r1 off;
+            sw  output r1 i3;
+
+            srl  r1 input l;
+            and  r1 r1 off;
+            sw  output r1 i4;
+
+            srl  r1 input m;
+            and  r1 r1 off;
+            sw  output r1 i5;
+
+            srl  r1 input n;
+            and  r1 r1 off;
+            sw  output r1 i6;
+
+            srl  r1 input o;
+            and  r1 r1 off;
+            sw  output r1 i7;
+
+            output: [u8; 8]
```

Comment:
It seems to me like everything except the SW instructions can be done outside assembly. Which I think would be clearer.

Reply:
If I have to calculate all the values first before inputting them into an assembly block, wouldnt that be worse performance wise?  Would use way more registers im guessing. Right now with no optimizations from the compiler the hash function im writing is gonna be very gas intensive so I need all the optimizations I can get

Reply:
Fully endorse the assembly if it actually uses more registers. But I'm not sure that would be the case here.